File: justsign_custom/public/py/item.py
import frappe

@frappe.whitelist()
def set_income_expense_accout_mandetory(doc, method=None):
    if doc.is_stock_item == 0 and doc.is_fixed_asset == 0 and not doc.item_defaults:
        frappe.throw("Please set Income Account and Expense Account in Item Defaults")

    for row in doc.item_defaults:
        if not row.income_account and not row.expense_account:
            frappe.throw("Please set Income Account or Expense Account in Item Defaults")

File: justsign_custom/public/js/item.js
frappe.ui.form.on('Item', {
	item_group(frm) {
		frappe.db.get_value("Item Group", frm.doc.item_group, "custom__is_service_group", (r) => {
            if (r && r.custom__is_service_group == 1) {
                frm.set_value("is_stock_item", 0);
                frm.set_df_property("is_stock_item", "read_only", 1);
                frm.set_df_property("is_fixed_asset", "read_only", 1);
            }             
	    }
    )}
})
